### ISSUE TITLE
Error when building using openssl configured with option no-ecdsa

### ISSUE THREAD

**Issue by Stefan-Ghinea** [libssh2/libssh2#549]
**Describe the bug**
Building libssh2 using openssl configured with option no-ecdsa leads to error:
../src/.libs/libssh2.so: undefined reference to `LIBSSH2_KEX_METHOD_EC_SHA_HASH_CREATE_VERIFY'

**To Reproduce**
e.g. on latest yocto:
```bash
git clone git://git.yoctoproject.org/poky.git
. poky/oe-init-build-env
echo "IMAGE_INSTALL_append = \" libssh2\"" > conf/local.conf
# add option no-ecdsa to DEPRECATED_CRYPTO_FLAGS in poky/meta/recipes-connectivity/openssl/openssl_1.1.1i.bb
bitbake libssh2
```

**Expected behavior**
libssh2 is built successfuly

**Comment by willco007** [comment]
To be clear, you have `OPENSSL_NO_ECDSA` but do have `LIBSSH2_ED25519` defined? It looks like that macro will need to be moved outside of the `LIBSSH2_ECDSA` check.

**Comment by Stefan-Ghinea** [comment]
Yes, this seems to be the case and moving the macro outside of the ```LIBSSH2_ECDSA``` check does solve the problem. (attached patch that does this)

[0001-kex.c.patch.zip](https://github.com/libssh2/libssh2/files/5831549/0001-kex.c.patch.zip)

**Comment by willco007** [comment]
should be resolved by #550.
